app/Http/Controllers/Api/UserController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\User\UpdateRequest;
+use App\Models\User;
+use App\Services\UserService;
+use Illuminate\Http\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+class UserController extends Controller
+{
+    public function __construct(private UserService $userService)
+    {
+    }
+
+    public function update(User $user, UpdateRequest $request): JsonResponse
+    {
+        if ($user->can('update', $user)) {
```

Comment:
we can move 'can' to the custom request, 'authorize' method

public function authorize()
    {
        return $this->user()->can('update', $this->route('user'));
    }